pkg/actuators/machine/actuator.go

```diff
@@ -0,0 +1,221 @@
+// Updating provider fields is not supported, a new machine should be created instead
+func (actuator *OvirtActuator) Update(ctx context.Context, machine *machinev1.Machine) error {
+	// eager update
+	providerSpec, err := ovirtconfigv1.ProviderSpecFromRawExtension(machine.Spec.ProviderSpec.Value)
+	if err != nil {
+		return actuator.handleMachineError(machine, "Update", apierrors.InvalidMachineConfiguration(
+			"Cannot unmarshal machineProviderSpec field: %v", err))
+	}
+
+	connection, err := actuator.getConnection()
+	if err != nil {
+		return actuator.handleMachineError(machine, "Update", apierrors.UpdateMachine(
+			"failed to create connection to oVirt API %v", err))
+	}
+
+	ovirtClient := ovirtC.NewOvirtClient(connection)
+	mScope := newMachineScope(ctx, ovirtClient, actuator.client, actuator.machinesClient, machine, providerSpec)
+	if err := mScope.patchMachine(ctx, conditionSuccess()); err != nil {
+		return actuator.handleMachineError(machine, "Update", apierrors.UpdateMachine(
+			"Error patching Machine %v", err))
+		return err
+	}
+
+	actuator.EventRecorder.Eventf(machine, corev1.EventTypeNormal, "Update", "Updated Machine %v", machine.Name)
+	return nil
+}
+
+// Update deletes the VM from the RHV environment
+func (actuator *OvirtActuator) Delete(ctx context.Context, machine *machinev1.Machine) error {
+	connection, err := actuator.getConnection()
+	if err != nil {
+		return err
+	}
+	ovirtClient := ovirtC.NewOvirtClient(connection)
+	mScope := newMachineScope(ctx, ovirtClient, actuator.client, actuator.machinesClient, machine, nil)
+	if err := mScope.delete(); err != nil {
+		return actuator.handleMachineError(machine, "Deleted", apierrors.UpdateMachine(
+			"error deleting Ovirt instance %v", err))
+		return err
+	}
+	actuator.EventRecorder.Eventf(machine, corev1.EventTypeNormal, "Deleted", "Deleted Machine %v", machine.Name)
+	return nil
+}
+
+// If the OvirtActuator has a client for updating Machine objects, this will set
+// the appropriate reason/message on the Machine.Status. If not, such as during
+// cluster installation, it will operate as a no-op. It also returns the
+// original error for convenience, so callers can do "return handleMachineError(...)".
+func (actuator *OvirtActuator) handleMachineError(machine *machinev1.Machine, reason string, err *apierrors.MachineError) error {
+	actuator.EventRecorder.Eventf(machine, corev1.EventTypeWarning, reason, "%v", err)
+
+	if actuator.client != nil {
+		machine.Status.ErrorReason = &err.Reason
+		machine.Status.ErrorMessage = &err.Message
+		if err := actuator.client.Update(context.TODO(), machine); err != nil {
+			return fmt.Errorf("unable to update machine status: %v", err)
+		}
+	}
+
+	klog.Errorf("Machine error %s: %v", machine.Name, err.Message)
+	return err
+}
+
+//getConnection returns a a client to oVirt's API endpoint
+func (actuator *OvirtActuator) getConnection() (*ovirtsdk.Connection, error) {
+	var err error
```

Comment:
You can add this variable declaration to the return type, making it a bit fewer lines.

Reply:
Done